tough/src/editor.rs

```diff
@@ -0,0 +1,269 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use crate::datastore::Datastore;
+use crate::error::{self, Result};
+use crate::key_source::KeySource;
+use crate::root_digest::{RootDigest, RootKeys};
+use crate::schema::{
+    Hashes, Role, RoleType, Root, Signature, Signed, Snapshot, SnapshotMeta, Target, Targets,
+    Timestamp, TimestampMeta,
+};
+use crate::transport::{TargetMapTransport, Transport};
+use crate::{Repository, Settings};
+use maplit::hashmap;
+use olpc_cjson::CanonicalFormatter;
+use ring::digest::{digest, SHA256, SHA256_OUTPUT_LEN};
+use ring::rand::{SecureRandom, SystemRandom};
+use serde::Serialize;
+use snafu::ResultExt;
+use std::collections::HashMap;
+use std::num::NonZeroU64;
+use std::path::{Path, PathBuf};
+use url::Url;
+
+/// An in-memory, unsigned TUF repository
+///
+/// This is provided as means to create and update a repository, which can then be signed.
+#[derive(Debug)]
+pub struct RepositoryEditor {
+    pub snapshot: Snapshot,
+    pub timestamp: Timestamp,
+    pub targets: Targets,
+    targets_meta: TargetMapTransport,
+}
+
+impl RepositoryEditor {
+    pub fn new(snapshot: Snapshot, timestamp: Timestamp, targets: Targets) -> RepositoryEditor {
```

Comment:
I don't know if this is a valid concern or not, but the only thing that seems strange to me is that you are taking the three known Role types as input. Maybe it should take the Repo itself as input? In the future there may be additional role types (https://github.com/awslabs/tough/issues/5).

And once the above observation occurs, then I wonder why these operations aren't mutable functions on a Repository itself?